foreman.te

```diff
@@ -446,14 +454,20 @@ corenet_tcp_bind_vnc_port(websockify_t)
 corenet_tcp_bind_vnc_port(foreman_rails_t)
 dev_read_urand(websockify_t)
 kernel_read_system_state(websockify_t)
-logging_send_syslog_msg(websockify_t)
+optional_policy(`
+    logging_send_syslog_msg(websockify_t)
+')
+optional_policy(`
+    abrt_stream_connect(websockify_t)
+')
 miscfiles_read_localization(websockify_t)
 sysnet_read_config(websockify_t)
-abrt_stream_connect(websockify_t)
 files_search_var_lib(websockify_t)
-read_files_pattern(websockify_t, puppet_var_lib_t, puppet_var_lib_t)
+optional_policy(`
+    read_files_pattern(websockify_t, puppet_var_lib_t, puppet_var_lib_t)
```

Comment:
It is not related to this PR, but puppet has `puppet_read_lib()`. It does not have a similar one for `puppet_etc_t` though anyway.

Reply:
Ok, there is `puppet_read_config()`, maybe too broad.

Reply:
Do you have an idea why it fails? It looks like it's on this line but these errors are usually never the case.

Reply:
Btw thanks I will refactor all of these, makes sense and makes it more cleaner, no need to require types explicitly now.

Reply:
I've never seen m4 statements in an optional block, try to use the interfaces.

Reply:
websockify should only need to read the Puppet certificate files. In Puppet AIO those are nowadays `puppet_etc_t` since they live in `/etc` now instead of `/var/lib/puppet`. We don't support non-AIO on EL7 & EL8 so that should be safe to update.

Reply:
Ok pushing.

Reply:
You were right it did not work previously, why? That's strange, I don't understand.